File: testing.rb
######################
##### BootCoder ######
######################
# Hunter T. Chapman  #
######################
##     TESTING      ##
######################



# INPUT SAMPLE

# Heelo Codevval | Hello Codeeval
# hELLO cODEEVAL | Hello Codeeval
# Hello Codeeval | Hello Codeeval

# OUTPUT SAMPLE
#
Low
# Critical
# Done

##########################
##  Initial   Solution  ##
##########################

def convert_bug_count(count)
  case count
  when (7..40)
    return "Critical"
  when (5..6)
    return "High"
  when (3..4)
    return "Medium"
  when (1..2)
    return "Low"
  when 0
    return "Done"
  end
end

File.open('input_sample.txt').each_line do |line|
# File.open(ARGV[0]).each_line do |line|
  line = line.chomp.split(" | ")
  coders = line[0].split(//)
  designers = line[1].split(//)
  count = 0
  coders.each_with_index do |element,index|
    count += 1 unless element == designers[index]
  end
  puts convert_bug_count(count)
end

File: testing.js
var displayBugCount = function(count){
  switch(true){
    case count > 6:
      return "Critical";
    case count > 4:
      return "High";
    case count > 2:
      return "Medium";
    case count > 0:
      return "Low";
    case count == 0:
      return "Done";
  };
};


var fs  = require("fs");
// fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
fs.readFileSync('input_sample.txt').toString().split('\n').forEach(function (line) {
    if (line != "") {

      var coders = line.split(" | ")[0];
      var designers = line.split(" | ")[1];
      var bugCount = 0;

      for(var i = 0; i<= coders.length; i += 1){
        if(coders[i] != designers[i]){
          bugCount += 1;
        };
      };

      console.log(displayBugCount(bugCount));

    };

});
